packages/library/package.json

```diff
@@ -73,12 +73,12 @@
         "@solana/keys": "workspace:*",
         "@solana/rpc-core": "workspace:*",
         "@solana/rpc-transport": "workspace:*",
-        "@solana/rpc-types": "workspace:*",
         "@solana/transactions": "workspace:*",
         "fast-stable-stringify": "^1.0.0"
     },
     "devDependencies": {
         "@solana/eslint-config-solana": "^1.0.2",
+        "@solana/rpc-types": "workspace:*",
```

Comment:
I like this change — i.e. using type-only dependencies as devDependencies — and the fact that it's consistent with the rest of the library. However, are we sure that this works for TypeScript end-users?

What I mean by that is: say I have package A which only needs types from package B and therefore imports it as a devDependency. Now a user installs package A and starts interacting with the types it provides. When it comes across a type defined in package B, does the user have that information? Or do they need to explicitly add package B as a devDependency as well?

The reason I'm asking this is, I initially wanted Umi to use the `Address` type of the new web3.js library when I refactored that bit. So I did exactly that, imported the `Address` type but imported `@solana/addresses` as a devDependency. Then, when I tested it in a sandbox application, TypeScript was complaining that the `Address` type had no definition.

I think in our case, it's less of an issue because the main library package exports `@solana/rpc-types` anyway so it's unlikely to be missing. I just have this internal fear that it's gonna fail for someone at some point and wanted to raise that so someone can reassure me haha.

Reply:
I am not the guy to do the reassuring here, but I will add another question!

If you have package B full of types, and inside your main package A you use none of the types, but you re-export B from A, does B tree-shake if a downstream user importing A doesn't use the types from B either?

Reply:
I'm 98.7% sure it does tree-shake re-exported items. Otherwise the main library — [which re-exports most of the sub-libraries](https://github.com/solana-labs/solana-web3.js/blob/62b6bd6381e231918ea8666922341c404ac03527/packages/library/src/index.ts#L1-L7) — would not be tree-shakeable at all and I can't imagine Steven letting that happen! 😅

Reply:
I'll move this back to a normal dependency in `library`, I don't think I meant to specifically change that one. I think I probably ran a `pnpm add --save-dev` without realising it was already a dependency and it got moved, or I accidentally moved it myself. I don't know exactly how these dependencies/re-exporting etc link together so I won't fiddle with it!

Reply:
@steveluscher ICYMI, I'd love your thoughts on the OP of this thread.

Reply:
My fears have been confirmed:

https://stackoverflow.com/a/46011417/11440277